tcpdf.php

```diff
@@ -20918,8 +20996,45 @@ protected function setPageBuffer($page, $data, $append=false) {
 	 */
 	protected function getPageBuffer($page) {
 		if (isset($this->pages[$page])) {
+			// Write last page to cache
+			if ($this->pageCacheFile !== null)
+			{
+				$lastPage = $page - 1;
+				if ($lastPage > 1 && isset($this->pages[$lastPage]))
+				{
+					// Duplicate file if needed
+					if ($this->roPageCacheFile)
+					{
+						$origPageCacheFile = $this->pageCacheFile;
+						self::$pageCacheRefCnts[(int)$this->pageCacheFile]--;
+						$this->pageCacheFile = fopen('php://temp', 'w+');
+						if ($this->pageCacheFile === false) {
+							$this->pageCacheFile = null;
+						} else {
+							self::$pageCacheRefCnts[(int)$this->pageCacheFile] = 1;
+							// Copy data
+							fseek($origPageCacheFile, 0, SEEK_SET);
+							while (($copyContent = fread($origPageCacheFile, 8192)) != false)
```

Comment:
This is missing {

Could you add a bit more comments on what each step does?

Reply:
I added the brace and added comments.